libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -2007,7 +2008,9 @@ std::optional<std::pair<PeriodData, std::vector<std::shared_ptr<Vote>>>> PbftMan
 
 blk_hash_t PbftManager::lastPbftBlockHashFromQueueOrChain() {
   auto pbft_block = sync_queue_.lastPbftBlock();
-  if (pbft_block) return pbft_block->getBlockHash();
+  if (pbft_block && pbft_block->getPeriod() >= getPbftPeriod()) {
```

Comment:
blocks in queue must be ordered by period right ? the last one has always the biggest period ?